### PR TITLE
7028 Intregrating Tailwind and applying it to the style-guide

### LINKED ISSUE #7037: 1 - Style Guide and Footer Component
This involves setting up Tailwind on the foundation site and using the new classes to recreate the Style Guide as well as the footer component.

This task should be reviewed by all devs if they agree with the move to tailwind before working on the epic further!

### PR DESCRIPTION
**Pull request by fessehaye**
Closes #7037 

Integrating tailwindCSS into the foundation site!
* Added tailwindCSS to our POSTCSS build and a few watch scripts on monitors html and JSX classes changes
* Replaced `network-api/networkapi/templates/partials/footer.html` with new tailwind Classes
* Replaced `/style-guide` as well

Most important thing to check is that there are no visual regressions on the style-guide or footer and that the dev enviornment still works as attended.

This is also devs opportunity to see how they like tailwinds use within the project as well.

Notes on how to play around with the Tailwind classes:
* https://tailwindcss.com/docs/editor-support Strongly recommend you install the tailwind intellsense plugin if you want to play around with the TW classes
Screen Shot 2021-07-26 at 4 57 07 PM

* If the plugin is not working try locally installing tailwind outside of docker by `npm install` from the root directory!
* All tailwind classes are prefixed with `tw-` to prevent conflicts with existing CSS and bootstrap.
* We are using the JIT interpreter so you can generate your own classes as well `ex:tw-text-[#34543f]` https://tailwindcss.com/docs/just-in-time-mode
* the buttons and dropdown are stilling using BT classes until we go and replace all of the buttons/form elements
* All tailwind configs are made in the `tailwind.config.js`
* You will see a mixed usage of applying the classes to the SASS files itself, Using the TW-classes directly on the templates, and creating some classes via the tailwind.config

* There is npm script included to reload the browser when files change using browser-sync, so you do not have to manually refresh all of the time!
  * You do have to have browser-sync globally installed `npm i -g browser-sync` and on a separate terminal run `npm run sync` while docker-compose is running.

### CODE REVIEW

**Review by sabrinang — CHANGES_REQUESTED** [review]
Thanks for updating the style guide Simon! I noticed a few things that could be updated:
- the boxes that contain the color swatch seems to have shifted inside or became thicker (it can match what we had previously)
- the $red color displays the blue color (change back to red)
- there a space missing after the title and parentheses for some gradient swatches
- change Design alias names for "phone" and "phone" to be "mobile (portrait)" and "mobile (landscape)"
- I noticed colors displayed as rgba values in the inspector, will we not have hex # values anymore?
- mobile view: color swatches look cleaner before taking up the whole width as it looks varied in the update (match what we had previously)
- mobile view: the breakpoint table is cut off on mobile (make it horizontally scrollable?)  

image

**Review by KalobTaulien — APPROVED** [review]
Code looks good to me. I like the `.tw-` prefix, that makes bootstrap conflicts less of a problem, so thank you for adding that in! 

I left a couple of code cleanup comments. But overall this is looking good. I'm going to give this an R+ as long as we address the code cleanup (in future tickets or now, I just don't want it to be forgotten). I still think we should have at least 2 devs review this though since this is a big thing moving forward.

User testing looks good for the footer. It looks like Sabrina has great feedback about the styleguide page which should be addressed before merging this in.

File: source/sass/type.scss [hunk 1]
```diff
@@ -22,77 +22,264 @@ a {
...
+      @apply tw-text-inherit tw-underline;
+    }
+  }
 }
 
 h4,
 .h4-heading {
-  @include heading("Zilla Slab", 400, 0, $black, $white);
-  @include set-text-size(22px, 28px, 24px, 30px);
+  @apply tw-font-normal tw-mb-4 tw-text-black tw-tracking-normal tw-font-zilla tw-text-[22px] tw-leading-[1.3] medium:tw-text-[24px];
```
[the remark this answers is not in the document]
Comment (fessehaye) [reply source/sass/type.scss:100]: Probably will refactor the headings another way in the future, either as components or part of a tailwind component plugin!

**Review by danielfmiranda — APPROVED** [review]
Hi Simon, thanks for tagging me in this! I think the code looks good, and like Kalob had mentioned earlier, I am a fan of the tw- prefix, as it makes it easier to see whether or not a component is using bootstrap classes still or tailwind. I am thinking this will be helpful when migrating in the future!

**Review by KalobTaulien — APPROVED** [review]

**Pull request merged**